app/src/Flashcards/index.js

```diff
@@ -5,9 +5,24 @@ import Flashcard from "./Flashcard";
 const Flashcards = ({ flashcards }) => {
   return (
     <>
-      {flashcards.map((flashcard) => {
-        return <Flashcard flashcard={flashcard} />;
-      })}
+      <h3>
+        These cards you haven't learnt yet. Total count:
+        {flashcards.filter((flashcard) => flashcard.is_learnt === false).length}
+      </h3>
+      {flashcards
+        .filter((flashcard) => flashcard.is_learnt === false)
+        .map((flashcard) => {
+          return <Flashcard flashcard={flashcard} />;
+        })}
+      <h3>
+        You have mastered these cards! Total count:
+        {flashcards.filter((flashcard) => flashcard.is_learnt === true).length}
+      </h3>
+      {flashcards
+        .filter((flashcard) => flashcard.is_learnt === true)
```

Comment:
I'd recommend separating the above logic (which is concerned with *which* data to present, i.e. filtering the data down) from the below logic (which is concerned with *how* to present the data, i.e. in `<Flashcard />` form).

You can do this by moving the logic for which data to show into two `useMemo` calls - one for deriving the flashcards that aren't yet learnt, and another one for deriving those that *are* learnt.  Then the *how* logic becomes a bit easier for you and others to read:

```
      {unlearntFlashcards
        .map((flashcard) => {
          return <Flashcard flashcard={flashcard} />;
        })}

      ...

      {learntFlashcards
        .map((flashcard) => {
          return <Flashcard flashcard={flashcard} />;
        })}
```